mono/utils/strenc.c

```diff
@@ -339,12 +339,17 @@ mono_utf8_validate_and_len_with_bounds (const gchar *source, glong max_bytes, gl
 
 		switch (*ptr) {
```

Comment:
Maybe we should make the complete switch statement a little more readable when changing it and make it follow code standard?

Reply:
Agreed.

Reply:
Let's fix this one later -- the warning and the formatting -- I removed this commit.

Reply:
The casts to guchar also aren't needed, and the two functions might be combinable.
The casts were probably that the pointer used to be to char but now is to guchar, so there is no truncation, it is comparing unsigned char to positive int.